enterprise/server/cmd/ci_runner/main.go

```diff
@@ -1858,7 +1858,9 @@ func (ws *workspace) fetch(ctx context.Context, remoteURL string, refs []string,
 	for _, filter := range *gitFetchFilters {
 		fetchArgs = append(fetchArgs, "--filter="+filter)
 	}
-	if fetchDepth > 0 {
+	if fetchDepth == 0 {
```

Comment:
just sanity checking, do we handle -1 ('smart' depth) correctly here?

Reply:
Yeah - we never directly pass the flag value, so we'll never pass -1. (Though I have considered how it's not ideal that we might accidentally pass -1. I wish it was a pointer and not an int)